intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp

```diff
@@ -0,0 +1,322 @@
+#include <windows.h>
+#include <signal.h>
+#endif
+
+std::shared_ptr<quant_layer_base> get_model_quant_layer(const std::string model_name) {
+  return ql_registry::create_ql(model_name);
+}
+
+class Model {
+ public:
+  Model() { model_init_backend(); }
+  ~Model() {
+    if (ctx) model_free(ctx);
+  }
+  void init_model(const std::string& model_path, int n_predict, int batch_size, int ctx_size, int seed, int threads,
+                  float repeat_penalty, const std::string& post_process);
+  void reinit();
+  std::string generate(const std::string& prompt, bool sentence_mode = true);
+  bool is_token_end() { return token_eos; }
+  static int quant_model(const std::string& model_path, const std::string& out_path, int bits, const std::string& alg,
+                         int block_size, const std::string& scale_dtype, const std::string& compute_type);
+
+ private:
+  model_context* ctx = nullptr;
+  gpt_params params;
+  std::vector<int> curr_input_ids;
+  int n_past = 0;
+  int n_vocab = 0;
+  int n_ctx = 0;
+  std::vector<model_token> last_n_tokens;
+  bool token_eos = false;
+
+  std::string generate_one_token(const std::string& prompt);
+  std::string generate_tokens(const std::string& prompt);
+  int post_process(float* logits);
+};
+
+void Model::init_model(const std::string& model_path, int max_new_tokens, int batch_size, int ctx_size, int seed,
+                       int threads, float repeat_penalty, const std::string& post_process) {
+#ifdef MODEL_NAME
+  params.model_name = MODEL_NAME;
+#endif
+  params.model_arch = model_name_to_arch::init().find(params.model_name);
+  params.model = model_path;
+  params.n_predict = max_new_tokens;
+  params.n_batch = batch_size;
+  params.n_ctx = ctx_size;
+  params.seed = seed;
+  params.n_threads = threads;
+  params.repeat_penalty = repeat_penalty;
+
+  n_past = 0;
+  token_eos = false;
+  curr_input_ids.clear();
+  ctx = model_init_from_gpt_params(params);
+  n_vocab = model_n_vocab(ctx);
+  n_ctx = model_n_ctx(ctx);
+  last_n_tokens.resize(n_ctx, 0);
+}
+
+void Model::reinit() {
+  n_past = 0;
+  last_n_tokens.resize(n_ctx, 0);
+  token_eos = false;
+  curr_input_ids.clear();
+}
+
+std::string Model::generate_one_token(const std::string& prompt) {
+  if (curr_input_ids.empty()) {
+    auto embd_inp = ::model_tokenize(ctx, prompt, false);
+    curr_input_ids = embd_inp;
+  }
+  for (auto item : curr_input_ids) {
+    last_n_tokens.erase(last_n_tokens.begin());
+    last_n_tokens.push_back(item);
+  }
+  model_eval(ctx, &curr_input_ids[0], curr_input_ids.size(), n_past, params.n_threads);
+  n_past += curr_input_ids.size();
+
+  float* logits = model_get_logits(ctx);
+  int next_token_id = post_process(logits);
+  curr_input_ids = {next_token_id};
+
+  if (next_token_id == model_token_eos() || n_past - prompt.size() == params.n_predict) {
+    token_eos = true;
+  }
+
+  auto next_token = model_token_to_str(ctx, next_token_id);
+  if (strcmp(next_token, "<|endoftext|>") == 0) {
```

Comment:
how about use `next_token_id == vocab.eos_token_id` for updating `token_eos` state?

Reply:
I check it here
```
  if (next_token_id == model_token_eos() || n_past - prompt.size() == params.n_predict) {
    token_eos = true;
  }
```
but some model will fail, so add endoftext checker

Reply:
`model_token_eos` is for `llama.cpp` which return fixed value. We add each model's special tokens ids  including `eos` in `vocab` struct.

Reply:
fixed, thx